src/pages/users/user-discount-request.component.js

```diff
@@ -0,0 +1,53 @@
+import Template from 'components/template/template.component';
+import useDiscountRequest from 'hooks/use-discount-request';
+import useCompanies from 'hooks/use-companies';
+import DiscountRequestTable from './components/discont-request-table.component';
+import useCompaniesSelectionMenuItem from 'hooks/use-companies-selection-menu-item';
+import { TextField, MenuItem } from '@material-ui/core';
+import { CircularProgress } from '@material-ui/core';
+
+const DiscountRequests = () => {
+  const { companies, isLoading: isLoadingCompanies } = useCompanies();
+
+  const { selectedCompany, handleChangeCompany } =
+    useCompaniesSelectionMenuItem(companies);
+
+  const { invoicesList, isLoading: isLoadingDiscountRequest } =
+    useDiscountRequest(selectedCompany);
+
+  const render = () => {
+    if (isLoadingCompanies)
+      return (
+        <Template>
+          <CircularProgress testid='company-edit-spinner' size={25} />
+        </Template>
+      );
```

Comment:
I like to avoid using curly braces only when I have an if statement in one line, for example:

```js
if (something) return somethingElse;
if (!isValid) return null;
```

In this specific case, the return has more than one line, so my suggestion is to use curly braces to improve the readability (this is a personal preference, feel free to disagree).

```suggestion
    if (isLoadingCompanies) {
      return (
        <Template>
          <CircularProgress testid='company-edit-spinner' size={25} />
        </Template>
      );
    }
```